Reorganize and update user setup section (#96)
jhudsl/AnVIL_Book_Instructor_Guide: 02_Instructor_Setup.Rmd

# (PART\*) Instructor Setup {-}

```{r, include = FALSE}
ottrpal::set_knitr_image_path()
```

# Overview 

<br>

The setup instructions are presented in two parts:

- **Instructor Setup:** covers everything needed to set up yourself and any co-instructors or TAs to work on AnVIL.  Depending on your team and funding, *you may only need to do these steps once* (and then make updates on an as-needed basis for changes in team members or funding).
- **Running the Class:** covers how to grant your students access to AnVIL.  These steps will need to be *repeated for each new offering of the course*.

## What do I need to do?

You may not need to do everything in the instructor setup section!

This section contains instructions for setting up billing from scratch.  If your funding is being managed by a third party (e.g. through a funding mechanism such as [STRIDES](https://datascience.nih.gov/strides) or through your institution) you can skip some steps.  The table below provides guidance on which steps you will need to complete:

> | Task | Self-managed Funding | 3rd-party Funding |

Reviewer: Love this - I like the idea of creating a table to help folks navigate our more involved or longer content

## Create a Google Account {#google-account}

Terra operates on Google Cloud Platform, so you will need a (free) Google account which will allow you to access the Terra platform to manage students and in-class analyses.

Your Google account will also (1) allow you to manage billing yourself or (2) allow a Program Manager to do so on your behalf.

```{r, child=c("_child_create_google_account.Rmd")}

```

## Sign in to Terra

You need to sign into Terra to allow Billing Project managers to add you to projects and/or Workspaces. [Launch Terra](https://anvil.terra.bio/#workspaces), and you should be prompted to sign in with your Google account.

You can always access Terra by going to [`anvil.terra.bio`](https://anvil.terra.bio/), or by clicking the link on the AnVIL home page.

```{r, echo=FALSE, fig.alt='Screenshot of the AnVIL home page. The button to "launch Terra" is highlighted.'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1tyt8ifxD3y7QW3nN27jh02aPrBvNEo1GN_u2Ne_UngM/edit#slide=id.gde7783227e_0_31")
```

> # Set up Google Billing (some users) {#google-billing-account}

Reviewer: I've been trying to avoid special characters - they break on leanpub/coursera renders

Author: oh, good to know, thanks

  repo_name = "jhudsl/AnVIL_Template"
)
```

### Add Users or Viewers (optional)

If you have a finance administrator who needs access to a Billing Account, you can add them with a few different levels of permissions.  Generally the most useful are:

- **Users** have a great deal of power over spending - they can create new "Billing Projects" and control who can spend money on those projects.  If you have an accounts manager responsible for expenses, it may make sense to add them as a Billing Account User.  If you wish to retain full control over who can spend money, you should not add any Users.
- **Viewers** can see the activity in the Billing Account but can’t make any changes.  This can be useful for finance staff who need access to the reports, or for anyone who needs to monitor spending.

```{r, echo = FALSE, results='asis'}
cow::borrow_chapter(
  doc_path = "child/_child_google_billing_add_member.Rmd",
  repo_name = "jhudsl/AnVIL_Template"
)
```

## Add Terra to Google Billing Account

```{r, echo = FALSE, results='asis'}
cow::borrow_chapter(
  doc_path = "child/_child_google_billing_add_terra.Rmd",
  repo_name = "jhudsl/AnVIL_Template"
)
```

> # Create Instructor/TA Group(s) {#instructor-group}

Reviewer: Same comment about special characters

However, an Admin for the Group can *indirectly* grant access to Terra resources.  By adding someone to the Group, they grant that person access to any Terra resources that the Group has been given permission to access.
:::

Often it's appropriate to add any **co-instructors as Admins, with TAs as Members**.

To add someone to your group:

:::: {.borrowed_chunk}
```{r, echo = FALSE, results='asis'}
cow::borrow_chapter(
  doc_path = "child/_child_terra_group_add_member.Rmd",
  repo_name = "jhudsl/AnVIL_Template"
)
```
::::

# Set up Terra Billing Projects {#terra-billing-projects}

Terra Billing Projects are how you grant people (including yourself!) permission to compute on AnVIL.  By adding someone to your Billing Project, you enable them to use your funding to carry out activities on AnVIL.

Depending on how you are being funded, this may be taken care of for you. If your funding is being managed by a third party (e.g. through a funding mechanism such as [STRIDES](https://datascience.nih.gov/strides), or through your institution), contact them to determine whether you should create Terra Billing Projects yourself.

> The first set of instructions below walk you through creating a Terra Billing Project.  If you do not need to create your own, you can skip down to [adding instructors to Billing Project](#add-instructors-billing-project).

Reviewer: ```suggestion
The first set of instructions below walk you through creating a Terra Billing Project.  **If someone else has provided a billing project**, you can skip down to [adding instructors to Billing Project](#add-instructors-billing-project).
```

Thoughts on rephrasing? Want to make it clear that unless the instructor is in communication with others, they will need to make their own.

### Add members to Billing Project

:::{.notice}
Users can be added to a Billing Project as either **Owners** or **Users**.  Owners can add and remove users from the Billing Project (including other Owners!).  This means Billing Project Owners have a great deal of power over how money can be spent.

It’s often a good idea to have at least one other Owner of a Billing Project in order to avoid getting locked out, in case the original Owner leaves or loses access to their account.  But you should exercise caution when selecing Billing Project Owners.

When teaching with a team, it may make sense to add only those most comfortable with Terra and/or most involved in Terra administration as Owners, and to add everyone else as Users.  This can be done by adding the Owners individually as Owners, and then adding the Instructor Group as User.
:::

:::: {.borrowed_chunk}
```{r, echo = FALSE, results='asis'}
cow::borrow_chapter(
  doc_path = "child/_child_terra_billing_project_add_member.Rmd",
  repo_name = "jhudsl/AnVIL_Template"
)
```
::::

# Wrap-up

**Congratulations! You have successfully set up AnVIL for your teaching team!**

> You and your team members should be free to develop and test content, funded by your Billing Project.  Checkout the *Content Preparation* section for more information about how to use or develop content on AnVIL.

Reviewer: ```suggestion
You and your team members should be free to develop and test content, funded by your Billing Project.  Check out the *Content Preparation* section for more information about how to use or develop content on AnVIL.
```
